### ISSUE TITLE
IntelliSense should not show myself when in base class/interface context

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#5399]
We shouldn't offer to implement/derive from ourselves:

![image](https://cloud.githubusercontent.com/assets/1103906/10034061/77ac231a-6142-11e5-96c5-a537fa9a42c0.png)

**Comment by Pilchie** [comment]
We should also remove: Enums, structs, and delegate types from the list here.

**Comment by CyrusNajmabadi** [comment]
> We shouldn't offer to implement/derive from ourselves:

Note: it's legal to write:

```
class C : C.Inner
{
    public interface Inner
    {

    }
}
```

We could consider filtering these out if there aren't available inner types the user might be referencing.

**Comment by CyrusNajmabadi** [comment]
@DoctorKrolic were you working on fixing this?

**Comment by DoctorKrolic** [comment]
Yes. The issue: https://github.com/dotnet/roslyn/issues/60935. And the PR: https://github.com/dotnet/roslyn/pull/62013. I'll get to it soon
